go-controller/pkg/ovn/pods.go

```diff
@@ -231,6 +231,52 @@ func (oc *Controller) waitForNodeLogicalSwitch(nodeName string) error {
 	return nil
 }
 
+func getPodAddresses(portName string) (net.HardwareAddr, net.IP, bool, error) {
+	podMac, podIP, err := util.GetPortAddresses(portName)
+	if err != nil {
+		return nil, nil, false, err
+	}
+	if podMac == nil || podIP == nil {
+		// wait longer
+		return nil, nil, false, nil
+	}
+	return podMac, podIP, true, nil
+}
+
+func waitForPodAddresses(portName string) (net.HardwareAddr, net.IP, error) {
+	var (
+		podMac net.HardwareAddr
+		podIP  net.IP
+		done   bool
+		err    error
+	)
+
+	// First try to get the pod addresses quickly using exponential backoff,
+	// then after about 2 seconds fall back to polling every second.
+	err = wait.ExponentialBackoff(
+		wait.Backoff{
+			Duration: 50 * time.Millisecond,
+			Steps:    4,
+			Factor:   2.1,
+			Jitter:   0.1},
```

Comment:
I think `Jitter` is only useful if you need to prevent multiple clients from all hitting the apiserver in unison, which wouldn't apply here since you wouldn't have multiple nodes reacting to the same pod creation event.

Reply:
Good point, can remove that.